x-pack/legacy/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -152,52 +156,42 @@ function buildDashboardUrlFromSettings(settings) {
           query = searchSourceData.query;
         }
 
-        // Add time settings to the global state URL parameter with $earliest$ and
-        // $latest$ tokens which get substituted for times around the time of the
-        // anomaly on which the URL will be run against.
-        const _g = rison.encode({
-          time: {
-            from: '$earliest$',
-            to: '$latest$',
-            mode: 'absolute',
-          },
-        });
-
-        const appState = {
-          filters,
-        };
-
-        // To put entities in filters section would involve creating parameters of the form
-        // filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b30fd340-efb4-11e7-a600-0f58b1422b87,
-        // key:airline,negate:!f,params:(query:AAL,type:phrase),type:phrase,value:AAL),query:(match:(airline:(query:AAL,type:phrase)))))
-        // which includes the ID of the index holding the field used in the filter.
-
-        // So for simplicity, put entities in the query, replacing any query which is there already.
-        // e.g. query:(language:kuery,query:'region:us-east-1%20and%20instance:i-20d061fa')
         const queryFromEntityFieldNames = buildAppStateQueryParam(queryFieldNames);
         if (queryFromEntityFieldNames !== undefined) {
           query = queryFromEntityFieldNames;
         }
 
-        if (query !== undefined) {
-          appState.query = query;
-        }
-
-        const _a = rison.encode(appState);
-
-        const urlValue = `kibana#/dashboard/${dashboardId}?_g=${_g}&_a=${_a}`;
-
-        const urlToAdd = {
-          url_name: settings.label,
-          url_value: urlValue,
-          time_range: TIME_RANGE_TYPE.AUTO,
-        };
-
-        if (settings.timeRange.type === TIME_RANGE_TYPE.INTERVAL) {
-          urlToAdd.time_range = settings.timeRange.interval;
-        }
+        const generator = npStart.plugins.directAccessLinks.getAccessLinkGenerator(
+          DASHBOARD_APP_LINK_GENERATOR
+        );
+
+        return generator
+          .createUrl({
```

Comment:
The URLs this is generating differ from the previous implementation in starting with `kibana#dashboard` where before it was `kibana#/dashboard`. Is this deliberate? If so, the code which generates the actual URL when drilling down from an anomaly, substituting in the times and entity values, will need amending as it checks for `kibana#/dashboard`, `kibana#/discover` (with the forward slash) - `x-pack/legacy/plugins/ml/public/application/util/custom_url_utils.ts`. Without the change, the earliest and latest times are not getting substituted on drilldown.

Reply:
ah interesting.  I guess there is no automated test coverage for the time substitution? Unless I missed the failed test. 

I'll see what I can do to put the `/` back.  I want to make these relative anyway since I realized that using the full url will cause a page refresh.

Reply:
The `/` should be back in latest commit, but I'm not sure how to test it.

Reply:
We have a number of unit tests in `x-pack/legacy/plugins/ml/public/application/util/custom_url_utils.test.ts` to test the substitutions, but these all work with hardcoded URLs. And our range of Jest tests and functional tests don't yet cover the complete workflow of creating the URL, and then testing the page that opens has all values substituted as expected.

The edit to put back the `/` LGTM, and when creating a custom URL off the edit job flyout on the ML Job Management page, you can hit the Test links to check the substitution is happening correctly.

![image](https://user-images.githubusercontent.com/7405507/75548144-74049000-5a24-11ea-9f9a-6cb0200fd481.png)

Reply:
@stacey-gammon testing your latest commits, I can confirm that the backslash `/` is now going in correctly.